Automattic/wp-calypso
# Domains: Create new redesigned transfer item components

### Changes proposed in this Pull Request

This PR adds the transfer item options to the new redesigned transfer page. Note: depends on PR #58261. Also note that this PR only encompasses the transfer item components, the other components in the page will be addressed in different PRs.

This is part of the domain management redesign project described in pcYYhz-m2-p2.

#### Screenshots

This is the old transfer page:

<img width="768" alt="Screen Shot 2021-11-22 at 12 58 31" src="https://user-images.githubusercontent.com/5324818/142894484-324a379d-cdc6-4700-b208-66637a0d0433.png">

And this is the new redesigned one:

<img width="1081" alt="Screen Shot 2021-11-22 at 12 56 02" src="https://user-images.githubusercontent.com/5324818/142894508-a9e0058e-9e0b-458e-b5f2-0f9847e2fcc4.png">

New page in mobile view:

<img width="397" alt="Screen Shot 2021-11-22 at 12 55 47" src="https://user-images.githubusercontent.com/5324818/142894531-6b1a710c-277e-4b02-9266-a516ec6cc19d.png">

### Testing instructions

- Build this branch locally or open the live Calypso link
- Go to "Upgrades > Domains"
- Select one of your registered domains
- If you're in the live Calypso link, please append `?flags=domains/transfers-redesign` to your URL to enable the appropriate feature flag
- Select the "Transfer your domain" options
- Ensure that a page with the "Transfer to another user" and "Transfer to another WordPress.com site" options are present and look like the screenshots above and the Figma project (hlVh2q24ad6MCwlwNnE9PQ-fi-600%3A52892)
- Verify that these two options take you to the correct pages
- Also please check that the mobile view looks correct
